physics/drag_suite.F90

```diff
@@ -1308,56 +1132,249 @@ subroutine drag_suite_run(                                           &
+               dusfc_bl(i)  = dusfc_bl(i) - onebgrcs * dtaux2d_bl(i,k) * del(i,k)
+               dvsfc_bl(i)  = dvsfc_bl(i) - onebgrcs * dtauy2d_bl(i,k) * del(i,k)
             enddo
          endif
 
       endif    ! if ( ls_taper(i) .GT. 1.E-02 )
 
    enddo   ! do i=its,im
 
-ENDIF  ! (do_gsl_drag_ls_bl).and.(gwd_opt_ls.EQ.1 .OR. gwd_opt_bl.EQ.1)
+ENDIF  ! (do_gsl_drag_ls_bl).and.(gwd_opt_ms .OR. gwd_opt_bl)
 
-if ( (gwd_opt == 33).or.(gwd_opt == 22) ) then
-  !  Finalize dusfc and dvsfc diagnostics
-  do i = its,im
-    dusfc_ls(i) = (-1./g*rcs) * dusfc_ls(i)
-    dvsfc_ls(i) = (-1./g*rcs) * dvsfc_ls(i)
-    dusfc_bl(i) = (-1./g*rcs) * dusfc_bl(i)
-    dvsfc_bl(i) = (-1./g*rcs) * dvsfc_bl(i)
-    dusfc_ss(i) = (-1./g*rcs) * dusfc_ss(i)
-    dvsfc_ss(i) = (-1./g*rcs) * dvsfc_ss(i)
-    dusfc_fd(i) = (-1./g*rcs) * dusfc_fd(i)
-    dvsfc_fd(i) = (-1./g*rcs) * dvsfc_fd(i)
-  enddo
+
+!====================================================================
+! Calculate small-scale gravity wave drag for stable boundary layer
+!====================================================================
+  XNBV=0.
+  tauwavex0=0.
+  tauwavey0=0.
+  density=1.2
+  utendwave=0.
+  vtendwave=0.
+!
+IF ( do_gsl_drag_ss ) THEN
+
+   do i=its,im
+
+      if ( ss_taper(i).GT.1.E-02 ) then
+   !
+   ! calculating potential temperature
+   !
+         do k = kts,km
+            thx(i,k) = t1(i,k)/prslk(i,k)
+         enddo
+   !
+         do k = kts,km
+            tvcon = (1.+fv*q1(i,k))
+            thvx(i,k) = thx(i,k)*tvcon
+         enddo
+
+         hpbl2 = hpbl(i)+10.
+         kpbl2 = kpbl(i)
+         !kvar = MIN(kpbl, k-level of var)
+         kvar = 1
+         do k=kts+1,MAX(kpbl(i),kts+1)
+!            IF (zl(i,k)>2.*var(i) .or. zl(i,k)>2*varmax) then
+            IF (zl(i,k)>300.) then
+               kpbl2 = k
+               IF (k == kpbl(i)) then
+                  hpbl2 = hpbl(i)+10.
+               ELSE
+                  hpbl2 = zl(i,k)+10.
+               ENDIF
+               exit
+            ENDIF
+         enddo
+         if((xland(i)-1.5).le.0. .and. 2.*varss_stoch(i).le.hpbl(i))then
+            if(br1(i).gt.0. .and. thvx(i,kpbl2)-thvx(i,kts) > 0.)then
+              ! Modify xlinv to represent wave number of "typical" small-scale topography 
+!              cleff_ss    = 3. * max(dx(i),cleff_ss)
+!              cleff_ss    = 10. * max(dxmax_ss,cleff_ss)
+!               cleff_ss    = 0.1 * 12000.
+              xlinv(i) = 0.001*pi   ! 2km horizontal wavelength
+              !govrth(i)=g/(0.5*(thvx(i,kpbl(i))+thvx(i,kts)))
+              govrth(i)=g/(0.5*(thvx(i,kpbl2)+thvx(i,kts)))
+              !XNBV=sqrt(govrth(i)*(thvx(i,kpbl(i))-thvx(i,kts))/hpbl(i))
+              XNBV=sqrt(govrth(i)*(thvx(i,kpbl2)-thvx(i,kts))/hpbl2)
+!
+              ! check for possibility of vertical wave propagation
+              ! (avoids division by zero if uwnd1(i,kpbl2).eq.0.)
+              if (uwnd1(i,kpbl2).eq.0.) then
```

Comment:
Is this section only executed when `ugwp_seq_update == T`? Otherwise, the comments in the declaration section are misleading and you can disregard my comment about not initializing them of course.

Reply:
No, it is always executed.  In the comments, note that the key word is "updated".  What it means is that the winds will be updated with the MSGWD+BLOCKING tendencies before use in the SSGWD+TOFD schemes only when ugwp_seq_update == T.  Otherwise, they will retain the original values of u1 and v1 passed in to the subroutine.

Reply:
I'll incorporate some of the words above into the comments of drag_suite.F90 so that it is more clear.